streamz/utils_test.py

```diff
@@ -125,3 +125,9 @@ def await_for(predicate, timeout, fail_func=None, period=0.001):
             if fail_func is not None:
                 fail_func()
             pytest.fail("condition not reached until %s seconds" % (timeout,))
+
+
+class metadata(Stream):
```

Comment:
I like this - could even be useful as a user class perhaps

Reply:
Yeah, I agree. Should that be in this PR or in another?

Reply:
Don't bother for now - in case someone asks :)